### PR TITLE
replace pkg/system Sequential funcs with moby/sys/sequential

### PR DESCRIPTION
**Pull request by thaJeztah**
- depends on https://github.com/moby/sys/pull/121

Migrating these functions to allow them being shared between moby, docker/cli,
and containerd, and to allow using them without importing all of sys / system,
which (in containerd) also depends on hcsshim and more.

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: vendor.mod [hunk 1]
```diff
@@ -56,6 +56,7 @@ require (
 	github.com/moby/swarmkit/v2 v2.0.0-20220721174824-48dd89375d0a
 	github.com/moby/sys/mount v0.3.3
 	github.com/moby/sys/mountinfo v0.6.2
+	github.com/moby/sys/sequential v0.0.0-20220829095930-b22ba8a69b30
```
Comment (thaJeztah) [inline vendor.mod:59]: No release yet, but I'll do a release once review of all PR's is looking good, and will update to `v0.5.0` (once release) in a follow-up.

**Comment by thaJeztah** [comment]
Ah, dang, needs a rebase after https://github.com/moby/moby/pull/44052 was merged. Could've seen that one coming 😂

**Comment by thaJeztah** [comment]
done 👍

**Review by AkihiroSuda — APPROVED** [review]

**Comment by thaJeztah** [comment]
@cpuguy83 this one LGTY as well?

**Review by cpuguy83 — APPROVED** [review]

**Pull request merged**